internal/component/prometheus/operator/configgen/config_gen_probe.go

```diff
@@ -71,6 +71,8 @@ func (cg *ConfigGenerator) GenerateProbeConfig(m *promopv1.Probe) (cfg *config.S
 		for k, v := range static.Labels {
 			grp.Labels[model.LabelName(k)] = model.LabelValue(v)
 		}
+		// Add namespace label
+		grp.Labels["namespace"] = model.LabelValue(m.Namespace)
```

Comment:
```suggestion
		// Add a namespace label if not already set. See https://github.com/prometheus-operator/prometheus-operator/commit/ffce8d01ff798135d5bbfcba1a88a87a418ee9fc.
		if _, ok := grp.Labels["namespace"]; !ok {
			grp.Labels["namespace"] = model.LabelValue(m.Namespace)
		}
```
The label should be set only if it wasn't already set by the the static labels. See the upstream code for details.